### DISCUSSION TITLE
Questions: How to resume a faulted workflow?

### DISCUSSION THREAD

**Discussion by buzzlu** [elsa-workflows/elsa-core#358] [Q&A]
Suppose a workflow has 3 steps.  The workflow was faulted at step 2  as program bugs. 

How can I resume this workflow from the step 2 after we fix  the program bug? 

Many thanks.

**Comment by sfmskywalker** [comment]
There are currently no programmatic means to resume a faulted workflow instance, but it's something we want to add. As a workaround, you could manipulate the workflow instance manually by updating the Status to Halted and BlockingActivities fields to an array with the activity ID of step 2.

**Reply by mrclayman** [reply]
Hi @sfmskywalker, thanks a lot for the tips. I am using Elsa 2 and right now, I am trying to follow your suggestions, but I am failing to make Elsa notice the change I make to the recovered workflow instance. I am wondering -- if I am to set the faulted workflow id to the blocking activity list, shouldn't I then signal Elsa somehow? Since blocking activities usually require a signal to wake up, it makes sense but I don't have a signal to send to the activity to restart it.

Also, there is no Halted WorkflowStatus value. Did you mean Suspended? (Or perhaps the instructions were for Elsa 1?) Just trying to clarify to make sure I am understanding correctly.

Thanks in advance for your response.

**Comment by buzzlu** [comment]
Got it. 

Many thanks.

**Comment by MehdiSaeedifar** [comment]
Hi, Is there any solution without manipulating the database?
